### PR TITLE
feat: Add edit sticker pack method.

### PR DESCRIPTION
**Pull request by kutuzov13**
# Description

Add method for editing a set of stickers.

# Checklist

- [ ] Linters and tests have passed
- [ ] I've written a changelog for PR change
- [ ] I'll make a release when PR is approved

### CODE REVIEW

File: botx/client/stickers_api/edit_sticker_pack.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        sticker_pack_id: UUID,
+        name: str,
+        preview: UUID,
+        sticker_order_id: Optional[List[UUID]],
+    ) -> "BotXAPIEditStickerPackRequestPayload":
+        return cls(
+            sticker_pack_id=sticker_pack_id,
+            name=name,
+            preview=preview,
+            stickers_order=sticker_order_id,
```
Comment (Nimond) [inline botx/client/stickers_api/edit_sticker_pack.py:29]: But `BotXAPIEditStickerPackRequestPayload` class doesn't have `stickers_order` field

File: botx/bot/bot.py [hunk 2]
```diff
@@ -1214,6 +1218,44 @@ async def get_sticker(
...
         return botx_api_sticker.to_domain()
 
+    async def edit_sticker_pack(
+        self,
+        *,
+        bot_id: UUID,
+        sticker_pack_id: UUID,
+        name: str,
+        preview: UUID,
+        sticker_order_id: List[UUID],
```
Comment (Nimond) [inline botx/bot/bot.py:1228]: Maybe name it `stickers_order`?

File: botx/bot/bot.py [hunk 3]
```diff
@@ -1214,6 +1218,44 @@ async def get_sticker(
...
+    ) -> StickerPack:
+        """Edit Sticker pack.
+
+        Args:
+            bot_id: Bot which should perform the request.
+            sticker_pack_id: Sticker pack id.
+            name: Sticker pack name.
+            preview: Sticker from the set selected as a preview.
+            sticker_order_id: List Sticker IDs in set in order they are displayed.
```
Comment (lxmnk) [inline botx/bot/bot.py:1237]: `Sticker IDs in order they are displayed`

File: botx/client/stickers_api/edit_sticker_pack.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            is_public=self.result.public,
+            stickers=[
+                Sticker(id=sticker.id, emoji=sticker.emoji, image_link=sticker.link)
+                for sticker in self.result.stickers
+            ],
+        )
+
+
+class EditStickerPackMethod(AuthorizedBotXMethod):
+    status_handlers = {
```
Comment (lxmnk) [inline botx/client/stickers_api/edit_sticker_pack.py:64]: You should use define `status_handlers` only if you want to override it.

File: botx/client/stickers_api/edit_sticker_pack.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        **AuthorizedBotXMethod.status_handlers,
+    }
+
+    async def execute(
+        self,
+        payload: BotXAPIEditStickerPackRequestPayload,
+    ) -> BotXAPIEditStickerPackResponsePayload:
+        jsonable_dict = payload.jsonable_dict()
+        sticker_pack_id = payload.jsonable_dict().pop("sticker_pack_id")
```
Comment (lxmnk) [inline botx/client/stickers_api/edit_sticker_pack.py:73]:
```
jsonable_dict = payload.jsonable_dict()
sticker_pack_id = jsonable_dict.pop("sticker_pack_id")
```

File: botx/client/stickers_api/edit_sticker_pack.py (hunk 1 above)
Comment (kutuzov13) [reply botx/client/stickers_api/edit_sticker_pack.py]: Fix.

File: botx/bot/bot.py (hunk 2 above)
Comment (kutuzov13) [reply botx/bot/bot.py]: Good idea. I corrected name.

File: botx/client/stickers_api/edit_sticker_pack.py (hunk 5 above)
Comment (kutuzov13) [reply botx/client/stickers_api/edit_sticker_pack.py]: Fix.

File: botx/bot/bot.py (hunk 3 above)
Comment (kutuzov13) [reply botx/bot/bot.py]: Thank you. Fix

File: botx/client/stickers_api/edit_sticker_pack.py (hunk 4 above)
Comment (kutuzov13) [reply botx/client/stickers_api/edit_sticker_pack.py]: Fix

File: botx/bot/bot.py [hunk 6]
```diff
@@ -1214,6 +1218,44 @@ async def get_sticker(
...
+
+        Args:
+            bot_id: Bot which should perform the request.
+            sticker_pack_id: Sticker pack id.
+            name: Sticker pack name.
+            preview: Sticker from the set selected as a preview.
+            stickers_order: Sticker IDs in order they are displayed.
+
+        Returns:
+            StickerPack.
```
Comment (lxmnk) [inline botx/bot/bot.py:1240]: Edited sticker pack.

Comment (kutuzov13) [reply botx/bot/bot.py]: Fix.

File: botx/client/stickers_api/edit_sticker_pack.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+
+class EditStickerPackMethod(AuthorizedBotXMethod):
+    async def execute(
+        self,
+        payload: BotXAPIEditStickerPackRequestPayload,
+    ) -> BotXAPIGetStickerPackResponsePayload:
+        jsonable_dict = payload.jsonable_dict()
+
+        path = f"/api/v3/botx/stickers/packs/{jsonable_dict['sticker_pack_id']}"
```
[the remark this answers is not in the document]
Comment (kutuzov13) [reply botx/client/stickers_api/edit_sticker_pack.py]: Fix.

**Review by Nimond — APPROVED** [review]

**Review by lxmnk — APPROVED** [review]

**Pull request merged**